src/formatDate.js

```diff
@@ -8,7 +8,26 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldDateFormat = date.split(fromFormat[fromFormat.length - 1]);
+  const newDateFormat = [];
+
+  for (let i = 0; i < oldDateFormat.length; i++) {
+    if (toFormat.indexOf(fromFormat[i]) === -1) {
```

Comment:
Using `indexOf` inside a loop for each iteration is inefficient because it traverses the `toFormat` array multiple times. Consider creating a map before the loop to quickly look up the index of each format element in `toFormat`.